exchanges/oex/oex.go

```diff
@@ -0,0 +1,522 @@
+		return resp, ErrorCapture(resp.ErrCapture.Error, resp.ErrCapture.Msg)
+	}
+
+	return resp, nil
+}
+
+// GetOpenOrders gets all the current delegation
+func (o *Oex) GetOpenOrders(symbol, pageSize, page string) (OpenOrderResponse, error) {
+	var resp OpenOrderResponse
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	params.Set("pageSize", pageSize)
+	params.Set("page", page)
+	path := o.APIUrl + oexAllOpenOrders
+	err := o.SendAuthHTTPRequest(http.MethodGet, path, params, &resp)
+	if err != nil {
+		return resp, err
+	}
+
+	if resp.Error != "0" {
+		return resp, ErrorCapture(resp.ErrCapture.Error, resp.ErrCapture.Msg)
+	}
+
+	return resp, nil
+}
+
+// SelfTrade stores information about self trades
+func (o *Oex) SelfTrade(side, orderType, volume, price, symbol, fee string) (SelfTradeResponse, error) {
+	var resp SelfTradeResponse
+	params := url.Values{}
+	params.Set("side", side)
+	params.Set("type", orderType)
+	params.Set("volume", volume)
+	params.Set("price", price)
+	params.Set("symbol", symbol)
+	params.Set("fee_is_user_exchange_coin", fee)
+	path := o.APIUrl + oexSelfTrade
+	err := o.SendAuthHTTPRequest(http.MethodPost, path, params, &resp)
+	if err != nil {
+		return resp, err
+	}
+
+	if resp.Error != "0" {
+		return resp, ErrorCapture(resp.ErrCapture.Error, resp.ErrCapture.Msg)
+	}
+
+	return resp, nil
+}
+
+// GetUserAssetData gets user asset and recharge data
+func (o *Oex) GetUserAssetData(uid, mobileNumber, email string) (UserAssetResponse, error) {
+	var resp UserAssetResponse
+	params := url.Values{}
+	params.Set("uid", uid)
+	params.Set("mobile_number", mobileNumber)
+	params.Set("email", email)
+	path := o.APIUrl + oexUserBalanceInfo
+	err := o.SendAuthHTTPRequest(http.MethodPost, path, params, &resp)
+	if err != nil {
+		return resp, err
+	}
+
+	if resp.Error != "0" {
+		return resp, ErrorCapture(resp.ErrCapture.Error, resp.ErrCapture.Msg)
+	}
+
+	return resp, nil
+}
+
+// FetchOrderInfo gets data of the order given its orderid
+func (o *Oex) FetchOrderInfo(orderID, symbol string) (FetchOrderResponse, error) {
+	var resp FetchOrderResponse
+	params := url.Values{}
+	params.Set("order_id", orderID)
+	params.Set("symbol", symbol)
+	path := o.APIUrl + oexUserBalanceInfo
+	err := o.SendAuthHTTPRequest(http.MethodPost, path, params, &resp)
+	if err != nil {
+		return resp, err
+	}
+
+	if resp.Error != "0" {
+		return resp, ErrorCapture(resp.ErrCapture.Error, resp.ErrCapture.Msg)
+	}
+
+	return resp, nil
+}
+
+// SendHTTPRequest sends an unauthenticated HTTP request
+func (o *Oex) SendHTTPRequest(path string, result interface{}) error {
+	var intermediary json.RawMessage
+	err := o.SendPayload(http.MethodGet,
+		path,
+		nil,
+		nil,
+		&intermediary,
+		false,
+		false,
+		o.Verbose,
+		o.HTTPDebugging,
+		o.HTTPRecording)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(intermediary, result)
+}
+
+// ErrorCapture deals with errors
+func ErrorCapture(code, message string) error {
+	return errors.New(message)
```

Comment:
code is unused. Can probably handle errors inside the functions to just return the err there